mysqld-exporter/main.libsonnet

```
@@ -27,7 +33,9 @@ local container = k.core.v1.container;
       + container.withPorts(k.core.v1.containerPort.new('http-metrics', 9104))
       + container.withArgsMixin([
         '--collect.info_schema.innodb_metrics',
-      ])
+      ] + if needsConfigFile then [
+        '--config.my-cnf=/conf/config.cnf',
+      ] else [])
```

Note that the suggestion might not be `jsonnetfmt` compliant. Wrapping the if/else block in parenthesis removes potential the _footgun_ about extending the code and only applying to one branch of the conditional block:

```suggestion
      ] + (if needsConfigFile then [
        '--config.my-cnf=/conf/config.cnf',
      ] else [])
      )
```